src/quantize/quantize.hpp

```diff
@@ -82,7 +82,7 @@ struct Quantize {
           const auto kernel = QuantizeKernel<input_t, output_t>{
               quantized_input_acc_t<input_t>{input, cgh},
               quantized_output_acc_t<output_t>{output, cgh}};
-          cgh.parallel_for(cl::sycl::range<1>{input.get_size()}, kernel);
```

Comment:
We usually pass the buffer in uint8_t (void* pointer style) and then apply the type inside the kernel. Unless you break this rule,  the get_count and get_size should not make any difference here.

Reply:
I don't think quantization will work with that at the moment, it relies on having a proper type. The tests and benchmarks seem to use typed buffers?